src/amoc_api_scenarios_handler.erl

```diff
@@ -10,112 +10,128 @@
+    {[binary()], cowboy:req(), state()}.
 allowed_methods(Req, State) ->
     {[<<"POST">>, <<"GET">>], Req, State}.
 
 -spec content_types_provided(cowboy:req(), state()) -> 
-        {[tuple()], cowboy:req(), state()}.
+    {[tuple()], cowboy:req(), state()}.
 content_types_provided(Req, State) ->
-    {[{<<"application/json">>, process_request}], Req, State}.
+    {[{<<"application/json">>, handle_get}], Req, State}.
 
 -spec content_types_accepted(cowboy:req(), state()) -> 
-        {[tuple()], cowboy:req(), state()}.
+    {[tuple()], cowboy:req(), state()}.
 content_types_accepted(Req, State) ->
-    {[{<<"application/json">>, process_request}], Req, State}.
-
--spec process_request(cowboy:req(), state()) -> 
-    {halt, cowboy:req(), state()}.
-process_request(Req0, State) ->
-    {ok, Data, Req1} = cowboy_req:body(Req0),
-    ContentType = {<<"content-type">>, <<"application/json">>},
-    try
-        Term = case jsx:is_json(Data) of
-            true -> jsx:decode(Data);
-            false -> undef
-        end,
-        {Status, Reply} = process_json(Term, State),
-        lager:info(Reply),
-        {ok, Req2} = cowboy_req:reply(Status, [ContentType], Reply, Req1),
-        {halt, Req2, State}
-    catch _:_ ->
-        ReplyE = jsx:encode([{<<"error">>, <<"unknown">>}]),
-        {ok, ReqE} = cowboy_req:reply(500, [ContentType], ReplyE, Req1),
-        {halt, ReqE, State}
-    end.
-
-%%%%%%%%%%%%%%
-% Request processing functions
-%%%%%%%%%%%%%%
--spec process_json(jsx:json_term(), state()) -> {integer(), jsx:json_text()}.
-process_json(Term, #state{action = start}) ->
-   ScenarioB = proplists:get_value(<<"scenario">>, Term),
-    Users = proplists:get_value(<<"users">>, Term),
-    
-    Scenario = erlang:binary_to_atom(ScenarioB, utf8),
-    
-    case code:load_file(Scenario) of
-        {error, nofile} ->
-            {500, jsx:encode([{<<"error">>, <<"module_not_exists">>}])};
-        {module, Scenario} -> 
-            _ = amoc_dist:do(Scenario, 1, Users),
-            {200, jsx:encode([{ScenarioB, <<"ok">>}])}
+    {[{<<"application/json">>, handle_post}], Req, State}.
+
+-spec malformed_request(cowboy:req(), state()) ->
+    {boolean(), cowboy:req(), state()}.
+malformed_request(Req, State) ->
+    {Method, Req2} = cowboy_req:method(Req),
+    malformed_request(Method, Req2, State).
+
+malformed_request(<<"POST">>, Req, State) ->
+    {ok, Body, Req2} = cowboy_req:body(Req),
+    case jsx:is_json(Body) of
+        false ->
+            {true, Req2, State};
+        true ->
+            JSON = jsx:decode(Body),
+            ContainModuleName = proplists:is_defined(<<"scenario">>, JSON),
+            ContainModuleSource = proplists:is_defined(<<"module_source">>, JSON),
+
+            case ContainModuleName and ContainModuleSource of
+                true ->
+                    ModuleName = proplists:get_value(<<"scenario">>, JSON),
+                    ModuleSource = proplists:get_value(<<"module_source">>, JSON),
```

Comment:
Line 80 is too long:                     ModuleSource = proplists:get_value(<<"module_source">>, JSON),.